[istio] Examples for redefine limit/request on sidecars in pod/deployment (#14211)
deckhouse/deckhouse: modules/110-istio/docs/EXAMPLES.md

{% alert level="warning" %}Available only in Enterprise Edition.{% endalert %}

To automate istio-sidecar upgrading, set a label `istio.deckhouse.io/auto-upgrade="true"` on the application `Namespace` or on the individual resources — `Deployment`, `DaemonSet` or `StatefulSet`.

> ## Customizing Istio Sidecar Resources

Reviewer: Why Sidecar capitalized? It is confusing with `Sidecar` custom resource.
```suggestion
## Customizing istio-proxy sidecar resource management
```

> You can override the global Istio sidecar resource limits for specific workloads by adding annotations to your Deployment, ReplicaSet, or Pod resources.

Reviewer: ```suggestion
You can override the global istio-proxy sidecar resource limits for specific workloads by adding annotations to your application Pods.
```

> ### Supported Annotations

Reviewer: ```suggestion
### Supported annotations
```

> Use these annotations to customize sidecar resources:

Reviewer: ```suggestion
Use these Pod annotations to customize sidecar resources:
```

| Annotation                          | Description                 | Example Value |
|-------------------------------------|-----------------------------|---------------|
| `sidecar.istio.io/proxyCPU`         | CPU request for sidecar     | `200m`        |
| `sidecar.istio.io/proxyCPULimit`    | CPU limit for sidecar       | `"1"`         |
| `sidecar.istio.io/proxyMemory`      | Memory request for sidecar  | `128Mi`       |
| `sidecar.istio.io/proxyMemoryLimit` | Memory limit for sidecar    | `512Mi`       |

### Configuration Examples

For Deployments:

```shell
apiVersion: apps/v1
kind: Deployment

> metadata:
>   annotations:
>     sidecar.istio.io/proxyCPU: 200m
>     sidecar.istio.io/proxyCPULimit: "1"
>     sidecar.istio.io/proxyMemory: 128Mi
>     sidecar.istio.io/proxyMemoryLimit: 512Mi
> # ... rest of your deployment spec

Reviewer: ```suggestion
metadata:
  ...
spec:
  template:
    metadata:
      annotations:
          sidecar.istio.io/proxyCPU: 200m
          sidecar.istio.io/proxyCPULimit: "1"
          sidecar.istio.io/proxyMemory: 128Mi
          sidecar.istio.io/proxyMemoryLimit: 512Mi
# ... rest of your deployment spec
```

The same for other app's controllers.
